src/sentry/tasks/post_process.py

```diff
@@ -274,22 +276,31 @@ def post_process_group(
                 ):
                     safe_execute(callback, event, futures)
 
-            has_commit_key = "workflow-owners-ingestion:org-{}-has-commits".format(
-                event.project.organization_id
-            )
-
             try:
-                org_has_commit = cache.get(has_commit_key)
-                if org_has_commit is None:
-                    org_has_commit = Commit.objects.filter(
-                        organization_id=event.project.organization_id
-                    ).exists()
-                    cache.set(has_commit_key, org_has_commit, 3600)
-
-                if org_has_commit and features.has(
-                    "projects:workflow-owners-ingestion", event.project,
-                ):
-                    process_suspect_commits(event=event)
+                lock = locks.get(
+                    "workflow-owners-ingestion:org-{}-has-group-lock".format(event.group_id),
+                    duration=10,
+                )
+                if not lock.locked():
```

Comment:
I realised I was wrong about this being necessary - it's fine to just use `.acquire` and rely on the exception being thrown.